integration/spark/src/main/scala/org/apache/spark/sql/optimizer/CarbonOptimizer.scala

```diff
@@ -449,12 +449,20 @@ class CarbonOptimizer(optimizer: Optimizer, conf: CatalystConf)
   // get the carbon relation from plan.
   def collectCarbonRelation(plan: LogicalPlan): Seq[CarbonDecoderRelation] = {
     plan collect {
-      case l@LogicalRelation(carbonRelation: CarbonDatasourceRelation, _) =>
-        CarbonDecoderRelation(l.attributeMap, carbonRelation)
+      case subQuery: Subquery if (isCarbonRelation(subQuery.child)) => subQuery.child match {
```

Comment:
Why it is changed this way? Any particular query is failing? Please provide more information.